Sources_v3/Controllers/MemberListController/MemberListController_Tests.swift

```diff
@@ -0,0 +1,525 @@
+                database: $0,
+                webSocketClient: $1,
+                apiClient: $2
+            )
+            return self.memberListUpdater!
+        },
+        memberListObserverBuilder: { [unowned self] in
+            self.memberListObserver = .init(
+                context: $0,
+                fetchRequest: $1,
+                itemCreator: $2,
+                fetchedResultsControllerType: $3
+            )
+            self.memberListObserver?.synchronizeError = self.memberListObserverSynchronizeError
+            return self.memberListObserver!
+        }
+    )
+}
+
+// A concrete `ChatChannelMemberListControllerDelegate` implementation allowing capturing the delegate calls
+private class TestDelegate: QueueAwareDelegate, ChatChannelMemberListControllerDelegate {
+    @Atomic var state: DataController.State?
+    @Atomic var didUpdateMembers_changes: [ListChange<ChatChannelMember>]?
+    
+    func controller(_ controller: DataController, didChangeState state: DataController.State) {
+        validateQueue()
+        self.state = state
+    }
+    
+    func memberListController(
+        _ controller: ChatChannelMemberListController,
+        didChangeMembers changes: [ListChange<ChatChannelMember>]
+    ) {
+        validateQueue()
+        didUpdateMembers_changes = changes
+    }
+}
+
+// A concrete `_ChatChannelMemberListControllerDelegate` implementation allowing capturing the delegate calls.
+private class TestDelegateGeneric: QueueAwareDelegate, _ChatChannelMemberListControllerDelegate {
+    @Atomic var state: DataController.State?
+    @Atomic var didUpdateMembers_changes: [ListChange<ChatChannelMember>]?
+    
+    func controller(_ controller: DataController, didChangeState state: DataController.State) {
+        validateQueue()
+        self.state = state
+    }
+    
+    func memberListController(
+        _ controller: ChatChannelMemberListController,
+        didChangeMembers changes: [ListChange<ChatChannelMember>]
+    ) {
+        validateQueue()
+        didUpdateMembers_changes = changes
+    }
+}
+
+extension ListChange {
+    var item: Item {
+        switch self {
+        case let .insert(item, at):
+            return item
+        case let .move(item, from, to):
+            return item
+        case let .remove(item, at):
+            return item
+        case let .update(item, at):
+            return item
+        }
+    }
+    
+    func fieldChange<Value>(_ path: KeyPath<Item, Value>) -> ListChange<Value> {
+        switch self {
+        case let .insert(item, at):
+            return .insert(item[keyPath: path], index: at)
+        case let .move(item, from, to):
+            return .move(item[keyPath: path], fromIndex: from, toIndex: to)
+        case let .remove(item, at):
+            return .remove(item[keyPath: path], index: at)
+        case let .update(item, at):
+            return .update(item[keyPath: path], index: at)
+        }
+    }
+}
```

Comment:
Why this?

Reply:
Good question! 

It's quite tricky to validate `[ListChange<Item>]` received by the delegate so I found those helpers useful but was too lazy to implement it in SDK target and write the tests. 

Do you think we should move this stuff outside test target?

Reply:
fixed in cc985467a6707399d7bbfa1d6b31f70134e6e0a1 and 3116d4f491d97e78646229d47352f7a98dbb217f